clippy_lints/src/iter_nth_zero.rs

```diff
@@ -0,0 +1,57 @@
+use crate::utils::span_lint_and_sugg;
+
+use rustc::declare_lint_pass;
+use rustc::lint::{EarlyContext, EarlyLintPass, LintArray, LintPass};
+use rustc_errors::Applicability;
+use rustc_session::declare_tool_lint;
+use syntax::ast::{Expr, ExprKind, PathSegment};
+
+declare_clippy_lint! {
+  /// **What it does:** Checks for the use of `iter.nth(0)`.
+  ///
+  /// **Why is this bad?** `iter.nth(0)` is unnecessary, and `iter.next()`
+  /// is more readable.
+  ///
+  /// **Known problems:** None.
+  ///
+  /// **Example:**
+  ///
+  /// ```rust,ignore
+  /// // Bad
+  ///  let mut s = HashSet::new();
+  ///  s.insert(1);
+  ///  let x = s.iter().nth(0);
+  ///
+  /// // Good
+  ///  let mut s = HashSet::new();
+  ///  s.insert(1);
+  ///  let x = s.iter().next();
+  /// ```
+  pub ITER_NTH_ZERO,
+  style,
+  "replace `iter.nth(0)` with `iter.next()`"
+}
+
+declare_lint_pass!(IterNthZero => [ITER_NTH_ZERO]);
+
+impl EarlyLintPass for IterNthZero {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &Expr) {
+        if let ExprKind::MethodCall(PathSegment { ident, .. }, args) = &expr.kind {
+            if ident.name.as_str() == "nth" {
```

Comment:
This would lint on any method call called `nth` which isn't what we want so we should check the type of the receiver to make sure it implements the `std::iter::Iterator` trait.

Reply:
Use `match_trait_method` in the `utils` module for an easy way doing this.

Reply:
good catch, thanks for pointing out that helper method